data/js/item/pumpkins.js

```diff
@@ -0,0 +1,324 @@
+// Other settings
+const randomizePumpkinCountdown = false; // If true, add/remove +1/-1 seconds to explosion potion countdowns
+
+function onCreateDFN( objMade, objType ) 
+{
+	var randomnamelist2006 = [
```

Comment:
Now thinking it might be a good time to solve this now ...

Reply:
The other option is to move SetRandomName off the CChar prototype for the JS engine to CBase, and alter the `SetRandomName` function to take a base object rather than char.

It'd be relatively trivial to do